src/core/main.c

```diff
@@ -1523,7 +1523,7 @@ int main(int argc, char *argv[]) {
                 /* Try to figure out if we can use colors with the console. No
                  * need to do that for user instances since they never log
                  * into the console. */
-                log_show_color(colors_enabled());
+                log_show_color(colors_enabled_no_tty(true));
                 make_null_stdio();
```

Comment:
so, SYSTEMD_COLORS= doesn't really make too much sense for PID 1. Maybe we should just check $TERM here, and that's it, i.e. call the suggested getenv_term_is_dumb() function here directly, and don't bother at all with checking for ttys, caching or SYSTEMD_COLORS or anything like that. I mean, PID 1 is special in many way, maybe we shouldn't try to shoehorn PID1's special color semantics into colors_enabled() at all...